cpp/bench/sg/kmeans.cu

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuML.hpp>
+#include <kmeans/kmeans.hpp>
+#include <utility>
+#include "benchmark.cuh"
+
+namespace ML {
+namespace Bench {
+namespace kmeans {
+
+struct Params {
+  DatasetParams data;
+  BlobsParams blobs;
+  ML::kmeans::KMeansParams kmeans;
+};
+
+template <typename D>
+class KMeans : public BlobsFixture<D> {
+ public:
+  KMeans(const std::string& name, const Params& p)
+    : BlobsFixture<D>(p.data, p.blobs), kParams(p.kmeans) {
+    this->SetName(name.c_str());
+  }
+
+ protected:
+  void runBenchmark(::benchmark::State& state) override {
+    if (!this->params.rowMajor) {
+      state.SkipWithError("KMeans only supports row-major inputs");
+    }
+    auto& handle = *this->handle;
+    auto stream = handle.getStream();
+    for (auto _ : state) {
+      CudaEventTimer timer(handle, state, true, stream);
+      ML::kmeans::fit_predict(handle, kParams, this->data.X, this->params.nrows,
+                              this->params.ncols, centroids, labels, inertia,
+                              nIter);
```

Comment:
Same here, we probably want to validate the accuracy of the outputs (it doesn't need to be in this PR, but I wanted to bring it up for discussion)

Reply:
issue #1083 should address this.